src/Sarif.Viewer.VisualStudio/Tags/SarifLocationTagger.cs

```diff
@@ -0,0 +1,573 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+namespace Microsoft.Sarif.Viewer.Tags
+{
+    using Microsoft.CodeAnalysis.Sarif;
+    using Microsoft.VisualStudio.Shell;
+    using Microsoft.VisualStudio.Text;
+    using Microsoft.VisualStudio.Text.Editor;
+    using Microsoft.VisualStudio.Text.Tagging;
+    using Microsoft.VisualStudio.TextManager.Interop;
+    using Microsoft.VisualStudio.Utilities;
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Linq;
+    using System.Threading;
+
+    internal class SarifLocationTagger : ITagger<TextMarkerTag>, ISarifLocationTagger, ITextViewCreationListener, IDisposable
+    {
+        /// <summary>
+        /// Protects access to the <see cref="SourceCodeFileToSarifTags"/> and <see cref="RunIdToSarifTags"/> dictionaries.
+        /// </summary>
+        private static ReaderWriterLockSlimWrapper TagListLock = new ReaderWriterLockSlimWrapper(new ReaderWriterLockSlim());
+
+        /// <summary>
+        /// Provides a dictionary from source code files (C/C++, etc.) to a set of tags to display in the VS editor.
+        /// </summary>
+        /// <remarks>
+        /// This is a static instance as a "tagger" is created based on an opened text buffer but the tags" persist beyond that instance
+        /// of a tagger and will be re-used if the text buffer is re-opened. (For example in a file close and re-open scenario).
+        /// </remarks>
+        private static readonly Dictionary<string, List<SarifLocationTag>> SourceCodeFileToSarifTags = new Dictionary<string, List<SarifLocationTag>>(StringComparer.OrdinalIgnoreCase);
+
+        /// <summary>
+        /// Provides a dictionary from SARIF log run Id to a set of tags to display in the VS editor.
```

Comment:
>og run Id  [](start = 46, length = 10)

Rename "run ID" to "log parse id" because SARIF logs can contain more than one run.

Reply:
But it still says "id" in the comment :)

---
In reply to: [482169104](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r482169104) [](ancestors = 482169104)